modules/collect/src/main/java/com/opengamma/collect/id/LinkResolutionException.java

```diff
@@ -18,21 +18,10 @@
   /**
    * Creates the exception passing the failed link.
    *
-   * @param link  the link which could not be resolved
-   * @param causeMessage  the reason why the link could not be resolved
+   * @param message  the reason why the link could not be resolved, null tolerant
    */
-  public LinkResolutionException(Link<?> link, String causeMessage) {
-    super("Unable to resolve link: " + link + " - " + String.valueOf(causeMessage));
-  }
-
-  /**
-   * Creates the exception passing the failed link.
-   *
-   * @param link  the link which could not be resolved
-   * @param cause  the reason why the link could not be resolved
-   */
-  public LinkResolutionException(Link<?> link, Failure cause) {
-    super("Unable to resolve link: " + link + " - " + (cause != null ? cause.getMessage() : "<no cause>"));
+  public LinkResolutionException(@Nullable String message) {
```

Comment:
Not sure about the change to this class. Aren't we realistically only going to get LinkResolutionExceptions from failed links? If so, why not allow the  link to form part of the exception?